packages/parser/test/macro-parse.test.ts

```diff
@@ -105,6 +107,106 @@ describe("Macro parsing", () => {
             $$(x-y.z{t})(m)
             $$f(x, y)`.split(/\n\s+/g);
 
+        for (const macroStr of validFunctions) {
+            it(`should print macro \`${macroStr}\``, () => {
+                const parsed = MacroParserV06.parse(macroStr);
+                expect(macroToStringV06(parsed)).toEqual(macroStr);
+            });
+        }
+    }
+});
+
+describe("Macro parsing of v0.7 macros", () => {
+    {
+        const validMacros = `$t
+            $t1
+            $_t
+            $t[1]
+            $t[1.5]
+            $t[$x]
+            $t[a][b]
+            $t.x
+            $t.x.y
+            $t.x[1].y
+            $t.x[1][2].y
+            $t.x[1].y[2]
+            $t{a}
+            $t{a="b"}
+            $t{a="$b"}
+            $t[1]{a="b"}
+            $t.x[1]{a="b"}.y
+            $(t)
+            $(x-y)
```

Comment:
Could you also add a test with `$x-y` with no parens to make sure that it gives the macro `$x` followed by "-y"?

Reply:
* `$(t)[1]`  (can break before `[` if want to )
* `$t[]`  (so eating empty [] is standardized)